core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/Eth2SignForIdentifierHandler.java

```diff
@@ -386,8 +384,8 @@ private String getAcceptableContentType(final List<MIMEHeader> mimeHeaders) {
   }
 
   private boolean isJsonCompatibleHeader(final MIMEHeader mimeHeader) {
-    final String mimeComponent = mimeHeader.component() + "/" + mimeHeader.subComponent();
-    return Objects.equals("application/json", mimeComponent)
-        || Objects.equals("*/*", mimeComponent);
+    final String mimeType =
+        mimeHeader.value(); // Must use value() rather than component() to ensure header is parsed
+    return Objects.equals("application/json", mimeType) || Objects.equals("*/*", mimeType);
```

Comment:
we may be able to enhance it by using `"application/json".equalsIgnoreCase(mimeType)"` to cover upper/lower or mixed case headers.

Reply:
done